zio-http/src/main/scala/zio/http/QueryParams.scala

```diff
@@ -72,16 +65,12 @@ final case class QueryParams(map: Map[String, Chunk[String]]) {
    */
   def encode(charset: Charset): String = QueryParamEncoding.default.encode("", self, charset)
 
-  override def equals(that: Any): Boolean = that match {
-    case that: QueryParams => self.normalize.map == that.normalize.map
-    case _                 => false
-  }
+  def equals(that: Any): Boolean
```

Comment:
This bears some discussion. Current implementation (now) implements equals as `this.toChunk == that.toChunk` and `.hashcode` as `this.toChunk.hashCode`. My initial inspection indicates that toChunk should be quite efficient,   given `Chunk.fromIterable` creates "Iterable backed chunks" rather than recreating the collection.

After getting all the wiring in place, I was able to scrub out `.map` entirely in favor of implemented `.getAll` directly against the underlying container. All of the mutations are now implemented in the trait as immutable transformations based on `.toChunk` -- recreating the `LinkedHashMap` for every change is necessary to work around it's mutability anyway, and it makes for a nice pure scala implementation.

This is a long way of saying what's the true and correct meaning of `.equals`? Given ordering is sometimes necessary, it seems an ordering-sensitive equality check is still appropriate, but I think the counter-argument can be made.

Reply:
Both `.equals` and `.hashCode` are implemented against `.normalize.toChunk` now.